tests/generic_relations/tests.py

```diff
@@ -513,6 +514,55 @@ def test_subclasses_with_parent_gen_rel(self):
         TaggedItem.objects.create(content_object=bear, tag='orange')
         self.assertEqual(Carrot.objects.get(tags__tag='orange'), bear)
 
+    def test_generic_inline_formsets_initial_count(self):
+        """
+        Test for initial form count considering save_as_new.
+        """
+        quartz = Mineral.objects.create(name="Quartz", hardness=7)
+
+        GenericFormSet = generic_inlineformset_factory(TaggedItem, extra=1)
+        data = {
+            'form-TOTAL_FORMS': '3',
+            'form-INITIAL_FORMS': '3',
+            'form-MAX_NUM_FORMS': '',
+        }
+        formset = GenericFormSet(data=data, prefix='form')
+        self.assertEqual(formset.initial_form_count(), 3)
+        formset = GenericFormSet(data=data, prefix='form', save_as_new=True)
+        self.assertEqual(formset.initial_form_count(), 0)
+
+    def test_generic_inline_formsets_save_as_new(self):
+        # The save_as_new parameter lets you re-associate the data to a new
+        # instance.  This is used in the admin for save_as functionality.
+        quartz = Mineral.objects.create(name="Quartz", hardness=7)
+        GenericFormSet = generic_inlineformset_factory(TaggedItem, extra=1)
+
+        # An immutable QueryDict simulates request.POST.
```

Comment:
I think a plain dict should be fine here. The `QueryDict` approach was because of a different issue (362fba87c9bb4f88542ba82ce4a732fed2634be7).